File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    static int result =0;
    public static int solution(int[] numbers, int target) {
        dfs(0,target,0,numbers);
        return result;
    }

    public static void dfs(int sum, int target, int depth, int[] numbers){
        
        if(depth == numbers.length){
            if(sum == target){
                result++;
           
            }
             return;
        }
            
            
       
        
        dfs(sum-numbers[depth],target, depth+1,numbers);
        dfs(sum+numbers[depth],target, depth+1,numbers);
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    

    def dfs(depth, total):
        if depth == len(numbers):
            if total == target:
                nonlocal answer
                answer += 1
            return
        if depth < len(numbers):
            dfs(depth+1, total+numbers[depth])
            dfs(depth+1, total-numbers[depth])
        
    dfs(0,0)
        
    return answer